app/src/main/java/com/concordia/mcga/utilities/pathfinding/GlobalPathFinder.java

```diff
@@ -114,6 +124,13 @@ private void sameBuildingNavigation() throws MCGAPathFindingException {
 
     }
 
+    private void setUpOutdoorDirections() {
+        outdoorDirections.setContext(activity);
+        outdoorDirections.setServerKey(activity.getResources().getString(R.string.google_maps_key));
+        outdoorDirections.setMap(map);
+        outdoorDirections.setSelectedTransportMode(transportMode);
+    }
```

Comment:
GlobalPathFinder should only be responsible for giving you lat,long. This is because shortest path computation cannot occur on the main thread. Interacting with outdoorDirections is out of the GlobalPathFinder's scope. This should be done on the UI (main) thread, as it interacts with the Google Map Fragment